### PR TITLE
Internalize `_CharacterClassModel`

### PR DESCRIPTION
**Pull request by hamishknight**
Rip out unused `_CharacterClassModel` API and make it an internal type by removing the last use of `makeDSLTreeCharacterClass`.

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

File: Sources/_StringProcessing/Regex/DSLTree.swift [hunk 1]
```diff
@@ -740,6 +740,32 @@ extension DSLTree {
     @_spi(RegexBuilder)
     public struct Atom {
       internal var ast: AST.Atom
+
+      // FIXME: The below APIs should be removed once the DSL tree has been
+      // migrated to use proper DSL atoms for them.
```
Comment (milseman) [inline Sources/_StringProcessing/Regex/DSLTree.swift:745]: What does this entail?

**Review by milseman — APPROVED** [review]
LGTM, thanks

File: Sources/_StringProcessing/Regex/DSLTree.swift (hunk 1 above)
Comment (hamishknight) [reply Sources/_StringProcessing/Regex/DSLTree.swift:745]: Adding them as dedicated cases to `DSLTree.Atom` (or splitting out a new `BuiltinCharacterClass` type perhaps), and then getting the consumer logic working with them. It's probably not too effort, but I'm going to leave it as a follow up for now. I can file an issue to track it.

Comment (hamishknight) [reply Sources/_StringProcessing/Regex/DSLTree.swift:745]: Filed https://github.com/apple/swift-experimental-string-processing/issues/579

**Pull request merged**